src/Routes/Devices/DeviceTable.js

```diff
@@ -276,8 +276,8 @@ const DeviceTable = ({
     : `${pathname}/systems`;
   const actionResolver = (rowData) => {
     const getUpdatePathname = (updateRowData) =>
-      historyProp
-        ? `/${updateRowData.rowInfo.id}/update`
+      navigateProp
+        ? `/insights/inventory/${updateRowData.rowInfo.id}/update`
```

Comment:
@acosferreira as we discussed, this might be causing an issue with the update navigation.
try these steps:
1- open the device with an update available
2- try to update
3- return to the device details by using the breadcrumb
4- try to update again

with these steps, the device will not redirect to the device update page.